packages/video_player/video_player_web/lib/video_player_web.dart

```diff
@@ -0,0 +1,196 @@
+import 'dart:async';
+import 'dart:html';
+import 'dart:ui' as ui;
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:video_player_platform_interface/video_player_platform_interface.dart';
+
+/// The web implementation of [VideoPlayerPlatform].
+///
+/// This class implements the `package:video_player` functionality for the web.
+class VideoPlayerPlugin extends VideoPlayerPlatform {
+  /// Registers this class as the default instance of [VideoPlayerPlatform].
+  static void registerWith(Registrar registrar) {
+    VideoPlayerPlatform.instance = VideoPlayerPlugin();
+  }
+
+  Map<int, _VideoPlayer> _videoPlayers = <int, _VideoPlayer>{};
+
+  int _textureCounter = 1;
+
+  @override
+  Future<void> init() async {
+    return _disposeAllPlayers();
+  }
+
+  @override
+  Future<void> dispose(int textureId) async {
+    _videoPlayers[textureId].dispose();
+    _videoPlayers.remove(textureId);
+    return null;
+  }
+
+  void _disposeAllPlayers() {
+    _videoPlayers
```

Comment:
nit: maybe `_videoPlayers.keys.forEach` would be a little nicer?

Reply:
Much nicer indeed.